### PR TITLE
docs(linter): add docs for config settings

### PR DESCRIPTION
**Pull request by DonIsaac**

### CODE REVIEW

**Comment by DonIsaac** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4827?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4827** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4827?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4826** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4826?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: tasks/website/src/linter/snapshots/schema_markdown.snap [hunk 1]
```diff
@@ -61,7 +61,9 @@ See [Oxlint Rules](./rules)
 
 type: `object`
 
-Shared settings for plugins
+Configure the behavior of linter plugins.
+
+## Example Here's an example if you're using Next.js in a monorepo: ```jsonc { "settings": { "next": { "rootDir": "apps/dashboard/" }, "react": { "linkComponents": [ // support next/link { "name": "Link", "linkAttribute": "to" } ] }, "jsx-a11y": { "components": { "Link": "a", "Button": "button" } } } } ```
```
Comment (DonIsaac) [inline tasks/website/src/linter/snapshots/schema_markdown.snap:66]: fixme

Comment (DonIsaac) [reply tasks/website/src/linter/snapshots/schema_markdown.snap]: This is problematic, it looks like JSONSchema does not preserve newlines. This is going to make supporting comments in JSON examples difficult.

**Comment by DonIsaac** [comment]
Blocked by https://github.com/GREsau/schemars/issues/313

**Comment by Boshen** [comment]
### Merge activity

* **Nov 21, 2:44 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Pull request merged**